flask/app/analyses.py

```diff
@@ -152,68 +152,74 @@ def get_analysis(id: int):
 def create_analysis():
     if not request.json:
         return "Request body must be JSON!", 415
-    try:
-        dts_pks = request.json["datasets"]
-    except KeyError:
-        return "No Dataset field provided", 400
-    try:
-        pipeline_pk = request.json["pipeline_id"]
-    except KeyError:
-        return "No Pipeline field provided", 400
 
-    if not dts_pks:
-        return "No Dataset IDs provided", 400
+    pipeline_id = request.json.get("pipeline_id")
+    if not isinstance(pipeline_id, int):
+        return "Missing pipeline_id field or invalid type", 400
 
-    if not pipeline_pk:
-        return "No Pipeline ID provided", 400
+    datasets = request.json.get("datasets")
+    if not (isinstance(datasets, list) and len(datasets)):
+        return "Missing datasets field or invalid type", 400
 
-    pipeline_id = models.Pipeline.query.get(pipeline_pk).pipeline_id
+    if not models.Pipeline.query.get(pipeline_id):
+        return "Pipeline not found", 404
 
-    now = datetime.now()
-    requested = now
-    updated = now
-    analysis_state = "Requested"
-    try:
-        requester_id = updated_by_id = current_user.user_id
-    except:  # LOGIN DISABLED
-        requester_id = updated_by_id = 1
+    if app.config.get("LOGIN_DISABLED"):
+        user_id = request.args.get("user")
+        requester_id = updated_by_id = user_id or 1
+    elif current_user.is_admin:
+        user_id = request.args.get("user")
+        requester_id = updated_by_id = user_id or current_user.user_id
+    else:
+        requester_id = updated_by_id = user_id = current_user.user_id
+
+    if user_id:
```

Comment:
Just noting that when I provide a non-existent user id with valid datasets (e.g. `curl -X POST -H "Content-Type: application/json" -d '{"datasets" : [7,8,9], "pipeline_id" : 2}' localhost:5000/api/analyses\?user=10`), I get the error `Some datasets were not found`. Not sure if this is an issue or not, as we would only be providing the user id when testing.

Reply:
This isn't an issue because it's pretty much only for testing with a potential extension in the future to actually support assuming identites.